cannon/mipsevm/iface.go

```diff
@@ -9,11 +9,23 @@ import (
 type FPVMState interface {
 	GetMemory() *memory.Memory
 
+	// GetHeap returns the current memory address at the top of the heap
+	GetHeap() uint32
+
+	// GetPreimageKey returns the most recently accessed preimage key
+	GetPreimageKey() common.Hash
+
+	// GetPreimageOffset returns the current offset into the current preimage
+	GetPreimageOffset() uint32
+
 	// GetPC returns the currently executing program counter
 	GetPC() uint32
 
-	// GetRegisters returns the currently active registers
-	GetRegisters() *[32]uint32
+	// GetCpu returns the currently active cpu scalars, including the program counter
+	GetCpu() CpuScalars
+
+	// GetRegistersMutable returns a pointer to the currently active registers
+	GetRegistersMutable() *[32]uint32
```

Comment:
Renamed to make it clearer that this returns a pointer and modifications to the returned array will mutate the underlying `FPVMState` registers.  Not sure if there are any existing naming conventions we use to distinguish whether a pointer is returned?

Reply:
GetRegistersRef is the convention.